### ISSUE TITLE
Columns Block: Allow for border options

### ISSUE THREAD

**Issue by davipontesblog** [WordPress/gutenberg#21889]
I don't find a way to set border styles on columns in the block editor unless custom CSS is applied. Here is an example of a custom CSS code to add borders between columns:

```
.wp-block-column {
    border-width: 0 1px 0 0;
    border-right-style: solid;
    border-right-color: black;
    padding-right: 35px;
}
```

We could allow users to select border styles as a block option in the editor. Additionally, some different border styles could be selectable, like "dotted", or "dashed" for example.

Thanks for considering this as an improvement to the Columns Block.

**Comment by karmatosed** [comment]
I am a bit torn as to if this should be added or is something for the table block to have. I would love other opinions on that though, so going to add a few labels to get wider input.

**Comment by alaczek** [comment]
YES! I strongly agree we need this!

Here are some visual examples of designs I had in mind. All of them are or could be built using columns block.

![image](https://user-images.githubusercontent.com/4550351/90201130-46b66d80-de1d-11ea-8cca-12e5b1ee781f.png)

![image](https://user-images.githubusercontent.com/4550351/90201637-c3961700-de1e-11ea-90d2-500892152156.png)

![image](https://user-images.githubusercontent.com/4550351/90202402-fccf8680-de20-11ea-995a-085a297b256d.png)

Also, similar idea was brought up in #21540)

> I am a bit torn as to if this should be added or is something for the table block to have.

I think it would be handy for table block, but columns block (and others like images or group) could really benefit from this feature too. It would be a shame if we had to use tables instead of columns, just because they allow for this particular styling option (although we would make a full-circle to the beginnings of web design where everything was a table :) )

**Comment by iamtakashi** [comment]
Border control would be a welcome addition to column block as well as other blocks like group, image, cover, and even query loop block. If there are many blocks that could benefit from border control, I think it makes sense to be worked on.

![Screenshot 2020-09-22 at 22 45 56](https://user-images.githubusercontent.com/908665/93941728-f6d59b00-fd26-11ea-894b-f04bace37d04.jpg)

**Comment by paaljoachim** [comment]
This can be very interesting! Border controls gives us additional design options.
I'll ping @ItsJonQ Jon for this issue, as he might have some thoughts to share here.
I will also add the Design Tools label.

**Comment by paaljoachim** [comment]
@aaronrobertshaw Aaron if I remember correctly, you are working on something like this.

**Comment by aaronrobertshaw** [comment]
Thanks for the mention @paaljoachim. 

Lately I've been working on adding [additional border block support features](https://github.com/WordPress/gutenberg/pull/30124). This looks like a good opportunity to take that for a spin. At the moment, the border block support does not allow configuring borders for individual sides however it is likely to evolve in that direction. It will just need some design input around the UI to achieve that.

Would it be an acceptable first step if you could only configure entire borders i.e. all sides at once? 

A theme could set `border: none` with enough specificity such that only the borders between columns would show. Maybe a block style could provide this ability as well.

When I get the bandwidth I'll create a proof of concept PR.

**Comment by paaljoachim** [comment]
Hey @aaronrobertshaw I happen to come across this issue as well.

Associated issue:
**Add gutter, margin, and padding controls to the columns block**
https://github.com/WordPress/gutenberg/issues/24874

**Comment by aaronrobertshaw** [comment]
> Hey @aaronrobertshaw I happen to come across this issue as well. 

Thanks for rounding all of these up! 

I have some pending changes to the borders block support UI that probably need to land before any PR addressing these issues can get approval. I'm hoping to make some progress on it in the coming week.

**Comment by aaronrobertshaw** [comment]
A proof of concept PR is up ( #31737 ). It might not cover every design eventuality but I think it should help for a start. Any feedback would be greatly appreciated. Thanks.

**Comment by aaronrobertshaw** [comment]
[Updated border support](https://github.com/WordPress/gutenberg/pull/37770) allowing for individual side border configuration has landed which unblocks the final PR adding border support to individual columns.

- https://github.com/WordPress/gutenberg/pull/39967

_Note: There is no responsive styling for individual column borders however this can be addressed via a follow-up issue._

**Comment by bradhogan** [comment]
@aaronrobertshaw Is the plan to allow for specific border changes as well (e.g. only add "border-left" or "border-right") versus having to wrap an entire block in the border?  This would be ideal to be able to create layouts like this?

![sample-borders](https://user-images.githubusercontent.com/1781226/163459265-bc4eb286-feee-449a-954f-173cf05a6eb7.png)

**Comment by aaronrobertshaw** [comment]
Hi @bradley2083 👋 

> Is the plan to allow for specific border changes as well (e.g. only add "border-left" or "border-right") versus having to wrap an entire block in the border?

Support for individual borders ( e.g. border-left or border-top etc. ) was added via https://github.com/WordPress/gutenberg/pull/37770 which merged about 5 days ago. 

If there is any functionality you feel is missing or there is something that could be refined, feel free to create an issue and I'll take a look.

https://user-images.githubusercontent.com/60436221/163895686-778036d4-cade-4e4f-826a-4c2d61f706ee.mp4

**Comment by bradhogan** [comment]
@aaronrobertshaw Nice, that looks perfect! Cheers.

**Comment by davipontesblog** [comment]
Great, thanks for implementing!

**Comment by BrunoAHVincent** [comment]
That works for parent columns, but what about child columns? I don't see that setting

![+borders](https://user-images.githubusercontent.com/59643134/176329152-0315d95a-d4ae-4b78-a73e-dba56dc6ab9f.jpg)

**Comment by aaronrobertshaw** [comment]
Hi @BrunoAHVincent 👋 

Borders for individual or child columns were added via https://github.com/WordPress/gutenberg/pull/39967 as the scope of this PR was limited only to the parent Columns block. Individual column border support didn't make it in time to land for WordPress 6.0 as it depended on new features that needed a little more time to stabilize. You can find more discussion in https://github.com/WordPress/gutenberg/pull/37770#issuecomment-1092747031.

The column borders you are chasing should land in 6.1 or are available if using the latest Gutenberg plugin.

Using the latest Gutenberg plugin, here's what I see.

https://user-images.githubusercontent.com/60436221/176330860-b4840ad4-4dbb-4d56-b38d-9d744553975b.mp4

I hope that helps 🙂

**Comment by BrunoAHVincent** [comment]
Excellent, thanks man, can do it with css for now, but good to know coming in next release

**Comment by bradhogan** [comment]
@aaronrobertshaw Any idea what happened to the individual border options (so you could add/manage top, bottom, left, and right borders individually)?  It no longer seems to be an option. 
<img width="294" alt="Screen Shot 2022-08-25 at 5 20 16 PM" src="https://user-images.githubusercontent.com/1781226/186785222-ff6268b6-e0b4-4147-b544-67410dc1f273.png">

**Comment by aaronrobertshaw** [comment]
@bradley2083 Individual borders are working ok for me. They've been in Gutenberg since [13.1](https://github.com/WordPress/gutenberg/releases/tag/v13.1.0), I believe. Which versions of WordPress and Gutenberg are you using? 

Here's what I'm currently seeing:

https://user-images.githubusercontent.com/60436221/186787038-e2bfa074-4b0e-40f8-9010-741033926c36.mp4

**Comment by bradhogan** [comment]
@aaronrobertshaw AH.  Somehow GB got deactivated on the site I'm working on.  Thanks for the quick reply, all good!

**Comment by aaronrobertshaw** [comment]
> Somehow GB got deactivated on the site I'm working on

Suspected it might be something along those lines. Glad it was easily resolved 👍 

> Thanks for the quick reply, all good!

Anytime!
